### ISSUE TITLE
CLI-998: `lando pull` failing with ACLI Acquia CLI 2.9.0.0

### ISSUE THREAD

**Issue by jds-tag** [acquia/cli#1428]
**Describe the bug**
As a developer, I want to be able to pull a database from Acquia via the Acquia Lando Recipe.

**To Reproduce**
On Acquia CLI 2.9.0.0 do the following:
1. Run `lando pull --code=none`
2. Select a DB
3. Files optional

You get this error message at the backup download phase:
```
        Downloading backup 241695962
PHP Fatal error:  Uncaught TypeError: Acquia\Cli\Command\Pull\PullCommandBase::getBackupPath(): Argument #2 ($database) must be of type AcquiaCloudApi\Response\DatabaseResponse, stdClass given, called in phar:///usr/local/bin/acli/src/Command/Pull/PullCommandBase.php on line 221 and defined in phar:///usr/local/bin/acli/src/Command/Pull/PullCommandBase.php:52
Stack trace:
#0 phar:///usr/local/bin/acli/src/Command/Pull/PullCommandBase.php(221): Acquia\Cli\Command\Pull\PullCommandBase::getBackupPath(Object(AcquiaCloudApi\Response\EnvironmentResponse), Object(stdClass), Object(AcquiaCloudApi\Response\BackupResponse))
#1 phar:///usr/local/bin/acli/src/Command/Pull/PullCommandBase.php(132): Acquia\Cli\Command\Pull\PullCommandBase->downloadDatabaseBackup(Object(AcquiaCloudApi\Response\EnvironmentResponse), Object(stdClass), Object(AcquiaCloudApi\Response\BackupResponse), Object(Closure))
#2 phar:///usr/local/bin/acli/src/Command/Pull/PullDatabaseCommand.php(50): Acquia\Cli\Command\Pull\PullCommandBase->pullDatabase(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput), false, false, false)
#3 phar:///usr/local/bin/acli/vendor/symfony/console/Command/Command.php(298): Acquia\Cli\Command\Pull\PullDatabaseCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#4 phar:///usr/local/bin/acli/src/Command/CommandBase.php(307): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#5 phar:///usr/local/bin/acli/vendor/symfony/console/Application.php(1058): Acquia\Cli\Command\CommandBase->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#6 phar:///usr/local/bin/acli/vendor/symfony/console/Application.php(301): Symfony\Component\Console\Application->doRunCommand(Object(Acquia\Cli\Command\Pull\PullDatabaseCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#7 phar:///usr/local/bin/acli/vendor/symfony/console/Application.php(171): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#8 phar:///usr/local/bin/acli/bin/acli(95): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#9 /usr/local/bin/acli(14): require('phar:///usr/loc...')
#10 {main}
  thrown in phar:///usr/local/bin/acli/src/Command/Pull/PullCommandBase.php on line 52

Fatal error: Uncaught TypeError: Acquia\Cli\Command\Pull\PullCommandBase::getBackupPath(): Argument #2 ($database) must be of type AcquiaCloudApi\Response\DatabaseResponse, stdClass given, called in phar:///usr/local/bin/acli/src/Command/Pull/PullCommandBase.php on line 221 and defined in phar:///usr/local/bin/acli/src/Command/Pull/PullCommandBase.php:52
Stack trace:
#0 phar:///usr/local/bin/acli/src/Command/Pull/PullCommandBase.php(221): Acquia\Cli\Command\Pull\PullCommandBase::getBackupPath(Object(AcquiaCloudApi\Response\EnvironmentResponse), Object(stdClass), Object(AcquiaCloudApi\Response\BackupResponse))
#1 phar:///usr/local/bin/acli/src/Command/Pull/PullCommandBase.php(132): Acquia\Cli\Command\Pull\PullCommandBase->downloadDatabaseBackup(Object(AcquiaCloudApi\Response\EnvironmentResponse), Object(stdClass), Object(AcquiaCloudApi\Response\BackupResponse), Object(Closure))
#2 phar:///usr/local/bin/acli/src/Command/Pull/PullDatabaseCommand.php(50): Acquia\Cli\Command\Pull\PullCommandBase->pullDatabase(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput), false, false, false)
#3 phar:///usr/local/bin/acli/vendor/symfony/console/Command/Command.php(298): Acquia\Cli\Command\Pull\PullDatabaseCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#4 phar:///usr/local/bin/acli/src/Command/CommandBase.php(307): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#5 phar:///usr/local/bin/acli/vendor/symfony/console/Application.php(1058): Acquia\Cli\Command\CommandBase->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#6 phar:///usr/local/bin/acli/vendor/symfony/console/Application.php(301): Symfony\Component\Console\Application->doRunCommand(Object(Acquia\Cli\Command\Pull\PullDatabaseCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#7 phar:///usr/local/bin/acli/vendor/symfony/console/Application.php(171): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#8 phar:///usr/local/bin/acli/bin/acli(95): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#9 /usr/local/bin/acli(14): require('phar:///usr/loc...')
#10 {main}
  thrown in phar:///usr/local/bin/acli/src/Command/Pull/PullCommandBase.php on line 52
```

**Expected behavior**
DB downloads and is imported with no issues

**Screenshots**
N/A, error message above.

**Desktop (please complete the following information):**
 - OS: Mac OS running Lando `v3.11.0`

**Additional context**
I was on ACLI `2.8`-ish (don't exactly remember) and upgraded bc other colleagues were having this issue. I was able to do a `lando pull` before the upgrade. Now I cannot.

**Comment by jds-tag** [comment]
Weird that since I posted this there is no longer a `2.9.0` release. This is still an issue with `2.8.6`. I downgraded to `2.8.0` and it works so there must be a breaking change somewhere between `2.8.0` and `2.8.6`.

**Comment by anavarre** [comment]
Hi @jds-tag - Sorry about the confusion around 2.9.0. It was published by mistake and was strictly identical to 2.8.6 FYI.

Thanks for reporting this issue. It is fixed via #1433 and will be available in 2.9.0. We hope to be able to release by EOW. If you'd rather not wait, feel free to temporarily use the Phar file from the #1433 build here: https://github.com/acquia/cli/actions/runs/4470700353/jobs/7854696680

<img width="679" alt="acli-phar" src="https://user-images.githubusercontent.com/293478/226432160-640394e1-d9b6-42df-a58e-4304fa93b314.png">

**Comment by anavarre** [comment]
We've been able to merge a number of PRs today. https://github.com/acquia/cli/releases/tag/2.9.0 is out early!

**Comment by anavarre** [comment]
https://github.com/acquia/cli/releases/tag/2.9.0 is released and fixes this issue via #1433

**Comment by Sen-Bounteous** [comment]
I was facing the same the issue with 2.8.3 and issue fixed in 2.9.2. Thanks 👍
